### ISSUE TITLE
Aurelia Dialog closes on mouse up not mouse down

### ISSUE THREAD

**Issue by radenkozec** [aurelia/dialog#275]
This is not a bug but not expected behavior.

**Current behavior:**
I have unexpected behavior on Aurelia Dialog. I have enabled auto-closing dialog.
What is the problem is that Aurelia Dialog captures events outside dialog on mouse up not mouse down event.
This way I have unwanted behavior when I have input box in Aurelia Dialog body with some text already populated.
User wants to select text and click mouse down, however user release mouse up outside boundaries of dialog. This cause dialog to close.
This happens quite often when input and other controls are near the edge of Aurelia Dialog.

**Expected/desired behavior:**
The way other dialog works for example if you look at Trello, It auto-close when you click outside dialog by capturing mouse down not mouse up event.
This prevents behavior I described above.

**Comment by StrahilKazlachev** [comment]
Currently the `click` event is used. I'll see what can be done to cover this use case, I also have one more in mind. Will follow up when I have something.

**Comment by RichiCoder1** [comment]
Ideally it would only close if both the mouse down & mouse up are outside the dialog bounds. That may be a little complicated however.

**Comment by PWKad** [comment]
Touch events, etc... also.  There are probably a number of references to use here for which events should be supported, Bootstrap modal always seems to have a well-documented list of events we could use to learn from probably.

**Comment by carlkenne** [comment]
This issue is something that I've run into as well. Any more thoughts on this?

**Comment by radenkozec** [comment]
@EisenbergEffect  @StrahilKazlachev  
Dialog should auto-close when you click outside by capturing **mouse down not mouse up** event.
Is there a any work done on this? 
I see you have major Dialog release but this issue still not fixed for 2 years. I suspect this is probably minor change in code?

**Comment by bigopon** [comment]
Shouldnt the events to listen to be configurable @StrahilKazlachev ? If so, maybe we can get some help @radenkozec

**Comment by radenkozec** [comment]
@bigopon If you don't have a time to fix this I maybe can try to help? I suspect this is one or few lines change.

**Comment by radenkozec** [comment]
@bigopon I have looked into code. I suspect it just needs to change 1 line of code in 

> dialog-renderer-default.ts

 on line 83 to change
` this.dialogLayout.addEventListener('click', this.contentOutboundClick);` to
` this.dialogLayout.addEventListener('mousedown', this.contentOutboundClick);`

I think this would solve this issue. Can someone check this or I can try to implement and test this?

**Comment by bigopon** [comment]
I think it's probably it. Maybe we just need to simply do a loop through a list of normalized event names to add listener to. Thanks for following this @radenkozec

**Comment by geertendoornenbal** [comment]
I tried this with version 1.1.0 (changing the event to mousedown), I needed to change the event on the dialogContainer, but also on the anchor. This is needed, so the event handler knows that for that same event the 'dialogHost' is clicked. Meaning the dialog won't be closed when clicked inside it.

This fixed the issue, I could start clicking inside the dialog, and stop clicking outside the dialog, without the dialog closing.

**Comment by d1vanloon** [comment]
I see that this issue has been closed, but I don't see a related PR or other resolution. What is the correct way to resolve this issue?

**Comment by EisenbergEffect** [comment]
Oops. That was an accidental close due to me misreading something. Apologies.

**Comment by LittleGnome** [comment]
I have created a pull request with the fixes as @geertendoornenbal described. Also tested it and everything works as expected.

**Comment by d1vanloon** [comment]
PR in question is #385 for reference.

**Comment by radenkozec** [comment]
@EisenbergEffect @LittleGnome @StrahilKazlachev @bigopon  I see you have new release but this fix is not included in it. Can you give us some update regarding this? We really need this fix.

**Comment by EisenbergEffect** [comment]
I've asked @bigopon and @StrahilKazlachev to review #385 to see if we can get it merged and released.

**Comment by radenkozec** [comment]
@LittleGnome @bigopon Can you confirm this is released in v2.0.0?
I get 2.0.0 and tried using it like this:
`this.dialogService.open({ viewModel: viewModel, model: editModel, mouseEventType: 'mouseup' })`
however dialog behaves exactly same like without this. 
Am I doing something wrong? Thanks

**Comment by LittleGnome** [comment]
@radenkozec The pull request was released in version 2.0.0 as far as I know. We are using it quiet some time by now. Only our settings contains `{ mouseEvent: "mousedown" }`.
While typing this line, I see that the geven property name in your configuration isn't correct. 
**mouseEventType** should be **mouseEvent**.

Default value is "click" and the other options are "mouseup" and "mousedown".

The reason why we called it _mouseEvent_ is to be consistent with the _keyEvent_ property and that one has also a typing that is called _KeyEventType_.

I hope this will help you out!
